runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -105,13 +107,20 @@ public void write(IToken token, Object data)
             l.getReplicationMode(token.getTokenValue())
                         .getReplicationProtocol(runtime)
                         .write(l, ld);
-
             return null;
          });
 
-
         // Cache the successful write
         if (!runtime.isCacheDisabled()) {
+            if (ld.hasCheckpointMetadata()) {
```

Comment:
I don't think this is necessary (and probably the wrong place to put if it is), the logdata should not have been serialized at this point, so any entry in the cache should reflect the correct in memory state already.